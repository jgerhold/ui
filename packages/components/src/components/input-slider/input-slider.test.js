import { mount } from '@vue/test-utils'
import InputSlider from './InputSlider'

describe('InputSlider', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(InputSlider)

      expect(wrapper.isVueInstance()).toBeTruthy()
    })
  })

  describe('props', () => {
    test('should render without props', async () => {
      const wrapper = await mount(InputSlider)
      expect(wrapper.element).toMatchSnapshot()
    })

    const props = [
      {
        prop: 'step',
        value: 1
      },
      {
        prop: 'value',
        value: 3
      },
      {
        prop: 'pins',
        value: [
          {
            value: 1,
            label: 'pin-label-1'
          },
          {
            value: 4,
            label: 'pin-label-2'
          }
        ]
      },
      {
        prop: 'background',
        value: '#fff'
      },
      {
        prop: 'borderColor',
        value: '#fff'
      }
    ]

    props.forEach(({ prop, value }) => {
      test(`should render type '${prop}'`, async () => {
        const wrapper = await mount(InputSlider, {
          propsData: {
            min: 1,
            max: 20,
            [prop]: value
          }
        })

        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })

  describe('events', () => {
    test('should dispatch input event on input', async () => {
      const wrapper = await mount(InputSlider, {
        propsData: {
          min: 1,
          max: 20
        }
      })

      wrapper.find('input').trigger('input')
      expect(wrapper.emitted('input')).toEqual([['0']])
    })

    test('should dispatch change event on change', async () => {
      const wrapper = await mount(InputSlider, {
        propsData: {
          min: 1,
          max: 20
        }
      })

      wrapper.find('input').trigger('change')
      expect(wrapper.emitted('change')).toEqual([['0']])
    })

    test('should dispatch dblclick event on dblclick', async () => {
      const wrapper = await mount(InputSlider, {
        propsData: {
          min: 1,
          max: 20
        }
      })

      wrapper.find('input').trigger('dblclick')
      expect(wrapper.emitted('dblclick')).toEqual([['0']])
    })
  })
})
