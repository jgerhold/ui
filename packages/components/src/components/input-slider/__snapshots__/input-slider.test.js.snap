// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSlider props should render type 'background' 1`] = `
<div
  class="input-slider"
>
  <input
    max="20"
    min="1"
    step="0.1"
    type="range"
  />
   
  <span
    class="track"
  />
    
  <span
    class="thumb"
    style="left: -5.2631578947368425%; background-color: rgb(255, 255, 255); border-color: #fff;"
  />
   
</div>
`;

exports[`InputSlider props should render type 'borderColor' 1`] = `
<div
  class="input-slider"
>
  <input
    max="20"
    min="1"
    step="0.1"
    type="range"
  />
   
  <span
    class="track"
  />
    
  <span
    class="thumb"
    style="left: -5.2631578947368425%; background-color: rgb(43, 138, 198); border-color: #fff;"
  />
   
</div>
`;

exports[`InputSlider props should render type 'pins' 1`] = `
<div
  class="input-slider"
>
  <input
    max="20"
    min="1"
    step="0.1"
    type="range"
  />
   
  <span
    class="track"
  />
   
  <span
    class="pin"
    style="left: 100%;"
  >
    
    pin-label-1
  
  </span>
  <span
    class="pin"
    style="left: 400%;"
  >
    
    pin-label-2
  
  </span>
   
  <span
    class="thumb"
    style="left: -5.2631578947368425%; background-color: rgb(43, 138, 198); border-color: #fff;"
  />
   
</div>
`;

exports[`InputSlider props should render type 'step' 1`] = `
<div
  class="input-slider"
>
  <input
    max="20"
    min="1"
    step="1"
    type="range"
  />
   
  <span
    class="track"
  />
    
  <span
    class="thumb"
    style="left: -5.2631578947368425%; background-color: rgb(43, 138, 198); border-color: #fff;"
  />
   
</div>
`;

exports[`InputSlider props should render type 'value' 1`] = `
<div
  class="input-slider"
>
  <input
    max="20"
    min="1"
    step="0.1"
    type="range"
  />
   
  <span
    class="track"
  />
    
  <span
    class="thumb"
    style="left: 10.526315789473685%; background-color: rgb(43, 138, 198); border-color: #fff;"
  />
   
</div>
`;

exports[`InputSlider props should render without props 1`] = `
<div
  class="input-slider"
>
  <input
    max="100"
    min="0"
    step="0.1"
    type="range"
  />
   
  <span
    class="track"
  />
    
  <span
    class="thumb"
    style="left: 0%; background-color: rgb(43, 138, 198); border-color: #fff;"
  />
   
</div>
`;
